### PR TITLE
Use MIT license for nullable annotations

### PR DESCRIPTION
**Pull request by drewnoakes**
These definitions are taken from dotnet/runtime (originally from dotnet/corefx) which uses the MIT license.

### CODE REVIEW

**Comment by davkean** [comment]
@drewnoakes You'll also need to include a 3rd party notice: https://github.com/dotnet/project-system/blob/dev15.9.x/THIRD-PARTY-NOTICES.md.

**Comment by drewnoakes** [comment]
@davkean I've added `THIRD-PARTY-NOTICES.txt`.

File: THIRD-PARTY-NOTICES.txt [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+The attached notices are provided for information only.
+
+
+License notice for dotnet/runtime
+---------------------------------
+
+https://github.com/dotnet/runtime/blob/master/LICENSE.TXT
+
+he MIT License (MIT)
```
Comment (davkean) [inline THIRD-PARTY-NOTICES.txt]:
```suggestion
The MIT License (MIT)
```

**Comment by drewnoakes** [comment]
We discussed this internally and agreed this was the right approach for now.

**Pull request merged**